src/main/scala/game/types/BoardMNSize.scala

```diff
@@ -0,0 +1,5 @@
+package game.types
+
+final case class BoardMNSize(m: Short, n: Short) {
```

Comment:
not sure if it makes sense...... 
probably it is ok to have in BoardMN the single separated value m and n...
it is just an over head. it is really never used except in BoardMN and after reused for the other derivatives.. probably better revert.